src/Build.OM.UnitTests/Definition/ProjectItem_Tests.cs

```diff
@@ -2171,6 +2172,79 @@ public void RemoveWithItemReferenceOnMatchingMetadata()
             items.ElementAt(1).GetMetadataValue("M2").ShouldBe("d");
         }
 
+        [Fact]
+        public void RemoveWithItemReferenceOnCaseInsensitiveMatchingMetadata()
+        {
+            string content = ObjectModelHelpers.FormatProjectContentsWithItemGroupFragment(
+                @"<I1 Include='a1' M1='1' M2='a'/>
+                <I1 Include='b1' M1='2' M2='x'/>
+                <I1 Include='c1' M1='3' M2='y'/>
+                <I1 Include='d1' M1='4' M2='b'/>
+
+                <I2 Include='a2' M1='x' m2='c'/>
+                <I2 Include='b2' M1='2' m2='x'/>
+                <I2 Include='c2' M1='3' m2='Y'/>
+                <I2 Include='d2' M1='y' m2='d'/>
+
+                <I2 Remove='@(I1)' MatchOnMetadata='m2' MatchOnMetadataOptions='CaseInsensitive' />");
+
+            var project = ObjectModelHelpers.CreateInMemoryProject(content);
+            var items = project.ItemsIgnoringCondition.Where(i => i.ItemType.Equals("I2"));
+
+            items.Select(i => i.EvaluatedInclude).ShouldBe(new[] { "a2", "d2" });
+
+            items.ElementAt(0).GetMetadataValue("M1").ShouldBe("x");
+            items.ElementAt(0).GetMetadataValue("M2").ShouldBe("c");
+            items.ElementAt(1).GetMetadataValue("M1").ShouldBe("y");
+            items.ElementAt(1).GetMetadataValue("M2").ShouldBe("d");
+        }
+
+        [Fact]
+        public void RemoveWithItemReferenceOnFilePathMatchingMetadata()
```

Comment:
Missing cases:
- relative path segments: `a/b/../c/./d` should match `a/c/d`
- relative path should match absolute path: when current directory is `c:\a` then `c:\a\b\c\` should match `b\c`. To test this you can use TestEnvironment to set the current directory for the duration of the test to a path that you hardcode in the project xml.